packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3864,7 +3865,23 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   }
 
   @override
-  RenderObject? findRenderObject() => renderObject;
+  RenderObject? findRenderObject() {
+    assert(() {
+      if (_lifecycleState != _ElementLifecycle.active) {
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary('Cannot get renderObject of inactive element.'),
+          ErrorDescription(
+            'In order for an element to have a valid renderObject, it must be '
+            'active, which means it is part of the tree.\n'
+            'Instead, this element is in the $_lifecycleState state.',
+          ),
+          describeElement('The findRenderObject() method was called for the following element'),
```

Comment:
I wonder if we should add some kind of a hint around "If you're calling this from a State object, consider guarding it with State.mounted." to the error to make this a little more actionable?

Reply:
Done